wgpu/src/image.rs

```diff
@@ -75,24 +82,25 @@ impl Pipeline {
             transform: Transformation::identity().into(),
         };
 
-        let uniforms_buffer = device.create_buffer_with_data(
-            uniforms.as_bytes(),
-            wgpu::BufferUsage::UNIFORM | wgpu::BufferUsage::COPY_DST,
-        );
+        let uniforms_buffer =
+            device.create_buffer_init(&wgpu::util::BufferInitDescriptor {
+                label: None,
+                contents: uniforms.as_bytes(),
+                usage: wgpu::BufferUsage::UNIFORM | wgpu::BufferUsage::COPY_DST,
```

Comment:
It seems a bit strange to create this one with contents, and with COPY_DST at the same time.
Since it's for uniforms, and they can change, wouldn't it be easier to create it *without* contents but then mark a flag somewhere that the initial contents are dirty, so that the rendering code will re-upload them?

Reply:
Fixed in 3eb63762c72ccd91717d4feb98128aa12cc1b126.